### PR TITLE
Helper function to get a valid topic name

### PR DESCRIPTION
**Pull request by chapulina**
Addresses https://github.com/ignitionrobotics/ign-gazebo/issues/239
Used by https://github.com/ignitionrobotics/ign-sensors/pull/33

Add a helper function to generate a valid topic name by converting spaces to `_` and stripping invalid character combinations.

### CODE REVIEW

File: src/TopicUtils.cc [hunk 1]
```diff
@@ -154,3 +155,25 @@ bool TopicUtils::DecomposeFullyQualifiedTopic(
...
+  std::string validTopic{_topic};
+
+  // Substitute spaces with _
+  validTopic = std::regex_replace(validTopic, std::regex(" "), "_");
+
+  // Remove special characters and combinations
+  validTopic = std::regex_replace(validTopic, std::regex("@"), "");
+  validTopic = std::regex_replace(validTopic, std::regex("~"), "");
+  validTopic = std::regex_replace(validTopic, std::regex("//"), "");
+  validTopic = std::regex_replace(validTopic, std::regex(":="), "");
```
Comment (iche033) [inline src/TopicUtils.cc:171]: I think we can merge these regex expressions:

```
validTopic = std::regex_replace(validTopic, std::regex("@|~|//|:="), "");
```

Comment (caguero) [reply src/TopicUtils.cc:171]: As a sanity check, can we measure how much time does it take to run this function? I remember seeing awful performance using `std::regex`.

Comment (chapulina) [reply src/TopicUtils.cc]: I combined all the regex calls in 43678c1.

---

_tl;dr regex performance does look bad, but I think it's acceptable for the current use case_

I did a little performance test comparing the time it takes to run the entire `asValidTopic` test 1000 times, without the `EXPECT` calls, in 2 different ways:

* Without changes to `AsValidTopic`: takes around 0.050 s
* With the regex calls commented out from `AsValidTopic`: takes around 0.0014 s

That does look like a lot. I'm not sure if there's a better alternative though, looping through the string feels prone to error. If all patterns were single characters I think we'd have more options.

For the usage we currently have on `ign-sensors`, I think this is acceptable though. This function is being called only once per sensor when their topic is set. Currently, this is not something that will be called repeatedly.

But let me know if you can think of a better alternative.

File: src/TopicUtils_TEST.cc [hunk 2]
```diff
@@ -242,6 +242,55 @@ TEST(TopicUtilsTest, testFullyQualifiedName)
...
+      "ABC/",
+      "/abc",
+      "/abc/d",
+      "/abc/d/e",
+      "a(bc)d-e_f=h+i.j"
+    })
+  {
+    auto valid = transport::TopicUtils::AsValidTopic(unmodified);
+    EXPECT_EQ(unmodified, valid);
+    EXPECT_TRUE(transport::TopicUtils::IsValidTopic(valid)) << valid;
```
Comment (caguero) [inline src/TopicUtils_TEST.cc:262]: I confess that I didn't know that the GTest macros were returning a stream. A good way to debug what failed I guess. :+1:

File: src/TopicUtils.cc (hunk 1 above)
Comment (caguero) [reply src/TopicUtils.cc]: It's totally acceptable to me as long as this is not called repeatedly. :+1:
